CONTRIBUTING.md

```diff
@@ -55,11 +55,15 @@ https://help.github.com/en/github/getting-started-with-github/git-and-github-lea
 
 3. Choose the correct branch to develop your changes against.
 
-   * Additions of new features to the codebase should be pushed to the `dev`
-     branch (`git checkout dev`).
+   * The `master` branch is kept in sync with the latest tagged release, but
+     should **not** be used as the base (i.e. target) branch of a pull request.
 
-   * Most bug fixes to previous (or frozen) releases should be made against the
-     corrisponding `release_XX.XX` branch (`git checkout release_XX.XX`).
+   * Additions of new features to the codebase should be based off the `dev`
+     branch (`git checkout -b feature_branch dev`), with few
+     [exceptions](doc/source/project/organization.rst#handling-pull-requests).
+
+   * Most bug fixes should target the oldest supported release exhibiting the
```

Comment:
💡Highlighting this fairly substantial change even though the wording did not change much. As I read it this is effectively requiring us to backport all bugs to the oldest supported release. I believe that until now we only had that for security-related bugs.

Reply:
I think this mostly as a clarification of what I intended before.
Any way I don't think it would require any backport from us, given this file is not part of our official procedures but a guide for contributors.

If you have suggestions for a better wording, I'm absolutely happy to make them in a follow-up, non-procedures PR. Works for you?

Reply:
I missed this is the non-mandatory file. Thanks. ✅